pkg/driver/trace_driver.go

```diff
@@ -100,13 +103,235 @@ func (d *Driver) runCSVWriter(records chan interface{}, filename string, writerD
 	writerDone.Done()
 }
 
-func DAGCreation(functions []*common.Function) *list.List {
+func createDAGWorkflow(functionList []*common.Function, function *common.Function, maxWidth int, maxDepth int) *list.List {
+	// Generating the number of Invocations per Depth
+	widthList := []int{}
+	widthList = append(widthList, 1)
+	for i := 1; i < maxDepth-1; i++ {
+		widthList = append(widthList, (rand.Intn(maxWidth-widthList[i-1]+1) + widthList[i-1]))
+	}
+	widthList = append(widthList, maxWidth)
+	queue := []*list.Element{}
 	linkedList := list.New()
-	// Assigning nodes one after another
-	for _, function := range functions {
-		linkedList.PushBack(function)
+	functionIndex := []int{}
+	for i := range functionList {
+		functionIndex = append(functionIndex, getName(functionList[i]))
+	}
+	functionId := getName(function)
+	functionIndex = append(functionIndex[:functionId], functionIndex[functionId+1:]...)
```

Comment:
May I kindly suggest adding some comments here and modifying the variable names? While I can understand the function's content, terms like 'queue' and 'linkedList' might be confusing for others trying to comprehend this code. Perhaps we could use more descriptive names that better reflect the purpose of these data structures in the context of this specific function.